pkg/cmd/infra/deployer/deployer.go

```diff
@@ -89,10 +91,10 @@ func NewDeployer(client kclient.Interface) *Deployer {
 		strategyFor: func(config *deployapi.DeploymentConfig) (strategy.DeploymentStrategy, error) {
 			switch config.Spec.Strategy.Type {
 			case deployapi.DeploymentStrategyTypeRecreate:
-				return recreate.NewRecreateDeploymentStrategy(client, kapi.Codecs.UniversalDecoder()), nil
+				return recreate.NewRecreateDeploymentStrategy(client, oclient, kapi.Codecs.UniversalDecoder()), nil
```

Comment:
@liggitt convinced me.  Casting the `oClient` down into the type we're expecting makes this easier to follow.